stubs/pluggy/pluggy/_hooks.pyi

```diff
@@ -0,0 +1,127 @@
+import sys
+from collections.abc import Callable, Generator, Mapping, Sequence, Set as AbstractSet
+from types import ModuleType
+from typing import Any, TypeVar, overload
+
+from ._result import _Result
+
+if sys.version_info >= (3, 8):
+    from typing import Final, TypedDict
+else:
+    from typing_extensions import Final, TypedDict
+
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extensions import TypeAlias
```

Comment:
We generally just import these from `typing_extensions` unconditionally. Type checkers think typing_extensions is part of the stdlib, so it doesn't have any impact on the dependencies of the stubs package if it's "unnecessarily" importing something from typing_extensions on Python 3.10, even though it "could" be importing the thing from typing:

```suggestion
from typing_extensions import Final, TypedDict, TypeAlias
```

Reply:
@AlexWaygood thank you! I remember now either you or another maintainer already pointed me at this once, but of course it didn't last long before I forgot about it :woman_facepalming: